CMakeLists.txt

```diff
@@ -0,0 +1,218 @@
+else()
+  set(ENV{CHPL_DEVELOPER} ${CHPL_DEVELOPER})
+  message(STATUS "CHPL_DEVELOPER updated to ${CHPL_DEVELOPER}")
+endif()
+
+# If we didn't get a value from the cmdline and the environment has a value, use it
+if ((NOT DEFINED ${CHPL_LLVM_GCC_PREFIX}) AND (NOT $ENV{CHPL_LLVM_GCC_PREFIX} STREQUAL ""))
+  message(NOTICE "CHPL_LLVM_GCC_PREFIX not set, using $ENV{CHPL_LLVM_GCC_PREFIX}")
+  set(CHPL_LLVM_GCC_PREFIX $ENV{CHPL_LLVM_GCC_PREFIX} CACHE STRING "Override CHPL_LLVM_GCC_PREFIX flag" FORCE)
+else()
+  set(ENV{CHPL_LLVM_GCC_PREFIX} ${CHPL_LLVM_GCC_PREFIX})
+  message(STATUS "CHPL_LLVM_GCC_PREFIX updated to ${CHPL_LLVM_GCC_PREFIX}")
+endif()
+
+
+# If any -DCHPL_ variables came to us from the command line, update them in the
+# environment so that they can be used by printchplenv
+if (DEFINED ${CHPL_HOME})
+  set(ENV{CHPL_HOME} ${CHPL_HOME})
+  message(NOTICE "CHPL_HOME updated to ${CHPL_HOME}")
+endif()
+if (DEFINED ${CHPL_HOST_PLATFORM})
+  set(ENV{CHPL_HOST_PLATFORM} ${CHPL_HOST_PLATFORM})
+  message(STATUS "CHPL_HOST_PLATFORM updated to ${CHPL_HOST_PLATFORM}")
+endif()
+if (DEFINED ${CHPL_HOST_COMPILER})
+  set(ENV{CHPL_HOST_COMPILER} ${CHPL_HOST_COMPILER})
+  message(STATUS "CHPL_HOST_COMPILER updated to ${CHPL_HOST_COMPILER}")
+endif()
+if (DEFINED ${CHPL_HOST_CC})
+  set(ENV{CHPL_HOST_CC} ${CHPL_HOST_CC})
+  message(STATUS "CHPL_HOST_CC updated to ${CHPL_HOST_CC}")
+endif()
+if (DEFINED ${CHPL_HOST_CXX})
+  set(ENV{CHPL_HOST_CXX} ${CHPL_HOST_CXX})
+  message(STATUS "CHPL_HOST_CXX updated to ${CHPL_HOST_CXX}")
+endif()
+if (DEFINED ${CHPL_HOST_ARCH})
+  set(ENV{CHPL_HOST_ARCH} ${CHPL_HOST_ARCH})
+  message(STATUS "CHPL_HOST_ARCH updated to ${CHPL_HOST_ARCH}")
+endif()
+if (DEFINED ${CHPL_HOST_MEM})
+  set(ENV{CHPL_HOST_MEM} ${CHPL_HOST_MEM})
+  message(STATUS "CHPL_HOST_MEM updated to ${CHPL_HOST_MEM}")
+endif()
+if (DEFINED ${CHPL_LLVM})
+  set(ENV{CHPL_LLVM} ${CHPL_LLVM})
+  message(STATUS "CHPL_LLVM updated to ${CHPL_LLVM}")
+endif()
+if (DEFINED ${CHPL_LLVM_SUPPORT})
+  set(ENV{CHPL_LLVM_SUPPORT} ${CHPL_LLVM_SUPPORT})
+  message(STATUS "CHPL_LLVM_SUPPORT updated to ${CHPL_LLVM_SUPPORT}")
+endif()
+if (DEFINED ${CHPL_LLVM_CONFIG})
+  set(ENV{CHPL_LLVM_CONFIG} ${CHPL_LLVM_CONFIG})
+  message(STATUS "CHPL_LLVM_CONFIG updated to ${CHPL_LLVM_CONFIG}")
+endif()
+if (DEFINED ${CHPL_SANITIZE})
+  set(ENV{CHPL_SANITIZE} ${CHPL_SANITIZE})
+  message(STATUS "CHPL_SANITIZE updated to ${CHPL_SANITIZE}")
+endif()
+
+# Makefile arguments we still want to support
+# TODO: SEE IF CMAKE COVERS THESE VARIABLES
+# TODO: Wire these up to do something
+set(CHPL_MAKE_ASSERTS OFF CACHE BOOL "ASSERTS flag from make")
+set(CHPL_MAKE_WARNINGS OFF CACHE BOOL "WARNINGS flag from make")
+set(CHPL_MAKE_OPTIMIZE OFF CACHE BOOL "OPTIMIZE flag from make")
+set(CHPL_MAKE_DEBUG OFF CACHE BOOL "DEBUG flag from make")
+set(CHPL_MAKE_PROFILE OFF CACHE BOOL "PROFILE flag from make")
```

Comment:
It doesn't make sense to me for these to have `CHPL_MAKE` prefix. Maybe they should just be `CHPL_ASSERTS` e.g., or perhaps just `ASSERTS`.